torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -234,18 +234,19 @@ void ProcessGroupAgent::sync() {
 void ProcessGroupAgent::start() {
   {
     std::lock_guard<std::mutex> futureLock{futureMutex_};
-    rpcRunning_.store(true);
+    rpcAgentRunning_.store(true);
```

Comment:
hmm, should this be set in `RpcAgent::start()` and move `RpcAgent::start` before kicking off the `futureTimeoutThread_ `?